contracts/adapters/nativeTokens/NativeTokenAdapter.sol

```diff
@@ -1,44 +1,63 @@
 // SPDX-License-Identifier: BUSL-1.1
 pragma solidity ^0.8.11;
 
-import "@openzeppelin/contracts/access/AccessControl.sol";
 import "../../interfaces/IBridge.sol";
 import "../../interfaces/IFeeHandler.sol";
 
 
-contract NativeTokenAdapter is AccessControl {
+contract NativeTokenAdapter {
     IBridge public immutable _bridge;
     bytes32 public immutable _resourceID;
 
-    event Withdrawal(address recipient, uint amount);
-
     error SenderNotAdmin();
     error InsufficientMsgValueAmount(uint256 amount);
     error MsgValueLowerThanFee(uint256 amount);
-    error TokenWithdrawalFailed();
     error InsufficientBalance();
     error FailedFundsTransfer();
-
-    modifier onlyAdmin() {
-        if (!hasRole(DEFAULT_ADMIN_ROLE, msg.sender)) revert SenderNotAdmin();
-        _;
-    }
+    error ZeroGas();
 
     constructor(address bridge, bytes32 resourceID) {
         _bridge = IBridge(bridge);
         _resourceID = resourceID;
-        _setupRole(DEFAULT_ADMIN_ROLE, msg.sender);
     }
 
     function deposit(uint8 destinationDomainID, string calldata recipientAddress) external payable {
-        if (msg.value <= 0) revert InsufficientMsgValueAmount(msg.value);
+        bytes memory depositData = abi.encodePacked(
+            uint256(bytes(recipientAddress).length),
+            recipientAddress
+        );
+        depositGeneral(destinationDomainID, depositData);
+    }
+
+    function depositToEVM(uint8 destinationDomainID, address recipient) external payable {
+        bytes memory depositData = abi.encodePacked(
+            uint256(20),
+            recipient
+        );
+        depositGeneral(destinationDomainID, depositData);
+    }
+
+    function depositToEVMWithMessage(uint8 destinationDomainID, address recipient, uint256 gas, bytes calldata message) external payable {
```

Comment:
` address recipient` is confusing here.
 Because if you provide the actual recipient address here, on the [execute proposal](https://github.com/sygmaprotocol/sygma-solidity/blob/204f13788802ca9eb0f8b5d24ae9689d1696ba5e/contracts/handlers/NativeTokenHandler.sol#L109)
 it will be picked as a DefaultMessageReceiver and will try to execute call `handleSygmaMessage`. Which doesn't make sense.
 Either we need to change executor logic or here need to pass zero address so executor then will pick defaultMessageReciever. 
 
 Passing zero address is confusing. So, we have two options.
 1. Change executor logic. In no way it should pick recipient.
 2. Remove` address recipient` param from this function and use zero address in `abi.encodePacked`

Reply:
> it will be picked as a DefaultMessageReceiver and will try to execute call handleSygmaMessage. Which doesn't make sense.

It will only try to execute handleSygmaMessage if there is an optional message included. Otherwise it will just transfer native token. This is needed for users who do not want to use default message receiver, and want to implement their own version of handleSygmaMessage.

Reply:
Added extensive description.

Reply:
Now, with comments, it is clear, and this way, we still have the flexibility to facilitate a broader scope of use cases. I will mark it as resolved @sadiq1971